### ISSUE TITLE
CTC loss outputs strange gradient

### ISSUE THREAD

**Issue by dmitriy-serdyuk** [SeanNaren/deepspeech.pytorch#121]
I modified the example of using the `CTCLoss` and discovered that something strange happens with the gradients:
```python
probs = torch.FloatTensor(
    [[[.9, .1], [.9, .1], [.9, .1], [.9, .1], [.9, .1], [.9, .1]]]
).transpose(0, 1).contiguous()
labels = Variable(torch.IntTensor([1]))
label_sizes = Variable(torch.IntTensor([1]))
probs_sizes = Variable(torch.IntTensor([6]))
probs = Variable(probs, requires_grad=True) # tells autograd to compute gradients for probs
probs.register_hook(lambda grad: setattr(probs, 'gradient', grad))
cost = ctc_loss(probs, labels, probs_sizes, label_sizes)
cost.backward()
print(probs.gradient.transpose(0, 1))
```
```
Variable containing:
(0 ,.,.) = 
 -0.1190  0.1190
 -0.0359  0.0359
 -0.0049  0.0049
 -0.0049  0.0049
 -0.0359  0.0359
 -0.1190  0.1190
[torch.FloatTensor of size 1x6x2]
```
So, basically, the gradient says that the probability of the class "0" should be increased despite the fact that the label is "1".

**Comment by dmitriy-serdyuk** [comment]
Some digging showed that the CTC expects linear outputs, not probabilities. I'm going to submit a PR fixing naming.
